### PR TITLE
Show enroll button on multi-group event if user is member of at least one group

### PR DESCRIPTION
**Pull request by zanvidmar**
Show enroll button on multi-group event if user is member of at least one group

Users are not able to join the multi-group event if they are not members of both groups, which is unexpected behaviour.
Changing the conceptual permission logic from "and" (user must be member of all groups) to "or" (user must be member of at least one group) fixes that issue.

## Problem
Situation:
- **Event 1**:
   - Group: “Group A” and “Group B”
   - Visibility: Members,
   - Registration method: Open registration
   - Unchecked: Allow Logins or Enrollments from Non-Group Members)
- **Group A**:
   - Members: 
      - User A
      - User C
- **Group B**:
   - Members:
      - User B
      - User C
- **User D**:
  - Not member of Group A and Group B

Facts:  
- User A can not enroll to Event 1 (bug)
- User B can not enroll to Event 1 (bug)
- User C can enroll to Event 1 (expected behaviour)
- User D can not enroll to Event 1 (expected behaviour)

## Solution
- User A should be able to see enroll button and enroll to Event 1
- User B should be able to see enroll button and enroll to Event 1
They both need to be able to see enroll button and to enroll to  Event 1 because they are both members of at least one group related to Event 1

## Issue tracker
[TB-6819](https://getopensocial.atlassian.net/browse/TB-6819)
[3281618](https://www.drupal.org/project/social/issues/3281618)

## Theme issue tracker
/

## How to test
- [ ] Set up the same "Situation" as described under "Problem"

QA 1:
- [ ] As UL User A (member of Group A) try to enroll the Event 1 (event related to Group A and Group B).
- [ ] It is expect that "Enroll" button is visible but it is not.
- [ ] The expected result to see the "Enroll" button is attained with this fix applied.

QA 2:
- [ ] As UL User B (member of Group B) try to enroll the Event 1 (event related to Group A and Group B).
- [ ] It is expect that "Enroll" button is visible but it is not.
- [ ] The expected result to see the "Enroll" button is attained with this fix applied.

QA 3:
- [ ] As UL User C (member of Group A and B) try to enroll the Event 1 (event related to Group A and Group B).
- [ ] It is expect that "Enroll" button is visible.
- [ ] The expected result to see the "Enroll" button with fix applied, confirms that fix did not break existing expected behaviour.

QA 4:
- [ ] As UL User D ( nor member of Group A or B) try to enroll the Event 1 (event related to Group A and Group B).
- [ ] It is expect that "Enroll" button is not visible nor user can even access the event.
- [ ] The expected result with fix applied, confirms that fix did not break existing expected behaviour.

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [x] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [x] If applicable (I.E. new hook_updates) the update path from previous versions (major and minor versions) are tested
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [x] New UI elements, or changes on UI elements are approved by the design team
- [x] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Release notes
Group members can now enroll to multi-group events if they are members of at least one of groups.

## Change Record
/

## Translations
/

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2941-5f6lkpqnbcn6zpnosap5yawzhqdtnq9h.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62879aa2a94aea8366e93c96

File: modules/social_features/social_event/src/Form/EnrollActionForm.php [hunk 1]
```diff
@@ -187,10 +193,21 @@ public function buildForm(array $form, FormStateInterface $form_state) {
             break;
```
Comment (zanvidmar) [inline modules/social_features/social_event/src/Form/EnrollActionForm.php:193]: It could be that this break is not properly handled for multi-group events and has potential to break the expected result. However this is out of scope of given bug.

File: modules/social_features/social_event/src/Form/EnrollActionForm.php [hunk 2]
```diff
@@ -187,10 +193,21 @@ public function buildForm(array $form, FormStateInterface $form_state) {
...
 
-          if ($group->hasPermission('enroll to events in groups', $current_user) == FALSE) {
-            return [];
+          if ($enroll_to_events_in_groups) {
+            // Skip permission validation if 'enroll to events in groups' is
+            // already granted.
+            break;
+          }
+          elseif ($group->hasPermission('enroll to events in groups', $current_user) === TRUE) {
+            $enroll_to_events_in_groups = TRUE;
```
Comment (NickDaelemans) [inline modules/social_features/social_event/src/Form/EnrollActionForm.php:202]: You can add a `break;` here. Then you can also remove the if statement above, since the foreach will be stopped, once you have set the `$enroll_to_events_in_groups = TRUE;`

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2941-5f6lkpqnbcn6zpnosap5yawzhqdtnq9h.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62879aa2a94aea8366e93c96

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2941-5f6lkpqnbcn6zpnosap5yawzhqdtnq9h.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62879aa2a94aea8366e93c96
